### PR TITLE
Enumerable methods

### PR DESCRIPTION
**Pull request by belsman**
### In this project we built a list of custom Enumerable methods that achieved similar functionality to those in the Ruby Enumerable documentation. The methods are as follows:

- #my_each similar to #each
- #my_each_with_index similar to  #each_with_index
- #my_select => #select
- #my_all? => #all?
- #my_any? => #any?
- #my_none? => #none?
- #my_count => #count
- #my_map => #map
- #my_inject => #inject

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+AllCops:
+  Exclude:
+    - "Guardfile"
+    - "Rakefile"
```
Comment (majovanilla) [inline .rubocop.yml:4]: Please make sure you use the Microverse rubocop file and don't change it.

File: README.md [hunk 2]
```diff
@@ -1 +1,92 @@
-# Enumerable-Methods
\ No newline at end of file
+# Enumerable methods
+
+> For this project, we created two **enumerable methods** used by the Enumerable module in Ruby.
+
```
Comment (majovanilla) [inline README.md]: Please update the information of the Readme.

File: my_enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return false if pattarn_or_cls.is_a?(Regexp) && el !~ pattarn_or_cls
+        return false if !pattarn_or_cls.is_a?(Regexp) && pattarn_or_cls.is_a?(Object) && !el.is_a?(pattarn_or_cls)
+      elsif block_given?
+        return false unless yield(el)
+      else
+        return false unless el
+      end
+    end
+    true
+  end
```
Comment (majovanilla) [inline my_enumerables.rb]: This method should work with Regex or a Class as a parameter.
Right now I get an error "Class or module required"
Example: `enumerable.my_all?(5)`

File: my_enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        return true if pattarn_or_cls.is_a?(Regexp) && el =~ pattarn_or_cls
+        return true if !pattarn_or_cls.is_a?(Regexp) && pattarn_or_cls.is_a?(Object) && el.is_a?(pattarn_or_cls)
+      elsif block_given?
+        return true if yield(el)
+      elsif el
+        return true
+      end
+    end
+    false
+  end
```
Comment (majovanilla) [inline my_enumerables.rb]: This method should work with Regex or a Class as a parameter.
Right now I get an error "Class or module required"
Example: `enumerable.my_any?(5)`
Example: `enumerable.my_any?('string')`

File: my_enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+  def my_none?(pattarn_or_cls = nil)
+    if !block_given?
+      !my_any?(pattarn_or_cls) { |el| el }
+    else
+      !my_any?(pattarn_or_cls) { |el| yield(el) }
+    end
+  end
```
Comment (majovanilla) [inline my_enumerables.rb:73]: This method should work with Regex or a Class as a parameter.
Right now I get an error "Class or module required"
Example: `enumerable.my_any?(5)`
Example: `enumerable.my_any?('string')`

File: my_enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    end
+    result
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/MethodLength
+  def my_inject(initial = nil, sym = nil)
+    result = to_a[0].is_a?(String) ? '' : 0
+    operator = nil
```
Comment (majovanilla) [inline my_enumerables.rb]: This method should work with ranges.
Example: `range.my_inject(:+)`

**Review by majovanilla — CHANGES_REQUESTED** [review]
# Changes required :mag: 

Hey team, 👋

## You did an *amazing*🌟 ✨ job on this project! 👊🏻  💯

You thought of almost all possible scenarios and your methods are passing most of the tests! That's awesome!

### There are still some changes 👇 you need to implement to approve this request 🔍

### Required Changes: ✍️

- Please check the comments I left on the code.


## Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack (@majo)
I would be happy to help! 👩‍💻 


Happy coding 🤓

--Maria Reyes

File: my_enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    end
+
+    if operator
+      my_each { |el| result = operations[operator].call(result, el) }
+    else
+      my_each { |el| result = yield(result, el) }
+    end
+
+    result
+  end
```
Comment (maelfosso) [inline my_enumerables.rb]:
```rb
2.6.5 :010 > array
 => [5, 5, 3, 1, 4, 2, 3, 8, 4, 2, 2, 8, 3, 8, 8, 4, 5, 7, 6, 7, 7, 7, 7, 5, 6, 6, 0, 2, 8, 8, 3, 4, 7, 1, 6, 5, 7, 4, 3, 1, 0, 6, 4, 8, 2, 1, 4, 8, 4, 8, 2, 1, 7, 0, 7, 1, 1, 8, 1, 8, 8, 0, 7, 8, 7, 5, 0, 0, 7, 3, 2, 4, 6, 8, 5, 8, 3, 8, 2, 6, 8, 5, 3, 0, 3, 5, 5, 6, 1, 2, 5, 8, 2, 2, 5, 3, 1, 7, 1, 6] 
2.6.5 :011 > operation = proc { |sum, n| sum + n }
2.6.5 :012 > array.my_inject(&operation)
 => 453 
2.6.5 :013 > array.my_inject(&operation)
 => 453 
2.6.5 :014 > array.inject(&operation)
 => 448 
2.6.5 :015 > array.inject(&operation)
 => 448 
2.6.5 :016 > 
```

Comment (maelfosso) [reply my_enumerables.rb]:
```rb
2.6.5 :027 > array
 => [5, 5, 3, 1, 4, 2, 3, 8, 4, 2, 2, 8, 3, 8, 8, 4, 5, 7, 6, 7, 7, 7, 7, 5, 6, 6, 0, 2, 8, 8, 3, 4, 7, 1, 6, 5, 7, 4, 3, 1, 0, 6, 4, 8, 2, 1, 4, 8, 4, 8, 2, 1, 7, 0, 7, 1, 1, 8, 1, 8, 8, 0, 7, 8, 7, 5, 0, 0, 7, 3, 2, 4, 6, 8, 5, 8, 3, 8, 2, 6, 8, 5, 3, 0, 3, 5, 5, 6, 1, 2, 5, 8, 2, 2, 5, 3, 1, 7, 1, 6] 
2.6.5 :028 > array.inject(:+)
 => 448 
2.6.5 :029 > array.my_inject(:+)
 => 453 
2.6.5 :030 > 

```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for applying the suggested changes 
However, there are some issues that need to be solve.
Kindly read the comments to learn more about them.
Once solved, kindly submit a new code review request.

Thanks

Mael FOSSO

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team  ,  you have done a brilliant job with this project. :clap:

#### You can merge !!!

**Pull request merged**